conda_forge_tick/migrators.xsh

```diff
@@ -368,3 +368,59 @@ class Compiler(Migrator):
 
     def remote_branch(self):
         return 'compiler_migration2'
+
+
+class Noarch(Migrator):
+    """Migrator for adding noarch."""
+    migrator_version = 0
+
+    compiler_pat = re.compile('.*_compiler_stub')
+    sel_pat = re.compile('(.+?)\s*(#.*)?\[([^\[\]]+)\](?(2)[^\(\)]*)$')
+    python_pat = re.compile('python( )?(?(1).*)')
+    unallowed_reqs = ['toolchain', 'gcc', 'cython', 'clangdev']
+
+    def filter(self, attrs):
+        conditional = (super().filter(attrs) or
+                       attrs.get('meta_yaml', {}).get('build', {}).get('noarch') or
+                       attrs.get('meta_yaml', {}).get('build', {}).get('noarch')
+                      )
+        if conditional:
+            return True
+        python = False
+        for req in attrs.get('req', []):
+            if self.compiler_pat.match(req) or req in self.unallowed_reqs:
+                return True
+            if self.python_pat.match(req):
+                python = True
+        if not python:
+            return True
+        for line in attrs.get('raw_meta_yaml', '').splitlines():
+            if self.sel_pat.match(line):
+                return True
+        return False
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        with indir(recipe_dir):
+            replace_in_file('build:', 'build:\n  noarch: python', 'meta.yaml', leading_whitespace=False)
+        return self.migrator_uid(attrs)
+
+    def pr_body(self):
+        body = super().pr_body()
+        body = body.format(
```

Comment:
Maybe include something about why the move to noarch? And maybe link to this https://conda-forge.org/docs/meta.html#building-noarch-packages ?